results/pkg/watcher/reconciler/controller.go

```diff
@@ -59,70 +67,152 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		r.logger.Errorf("invalid resource key: %s", key)
-		return nil
+		return err
+	}
+
+	pEMsg, prErr := r.reconcilePipelineRun(ctx, namespace, name)
+	tEMsg, trErr := r.reconcileTaskRun(ctx, namespace, name)
+
+	emsgs := []string{pEMsg, tEMsg}
+	emsg, err := getReconcileError(emsgs, prErr, trErr)
+	if err != nil {
+		r.logger.With(zap.String("namespace", namespace), zap.String("name", name)).Error(emsg)
 	}
+	return err
+}
+
+func isReconcileSuccess(errs ...error) bool {
+	for _, err := range errs {
+		if err == nil {
+			return true
+		}
+	}
+	return false
+}
+
+func getReconcileError(emsgs []string, errs ...error) (string, error) {
+	if isReconcileSuccess(errs...) {
+		return "", nil
+	}
+	for idx, err := range errs {
+		if !errors.IsNotFound(err) {
+			return emsgs[idx], err
+		}
+	}
+	return "Can't find valid Tekton Result resource", errors.NewNotFound(v1beta1.Resource("Tekton Result"), "Tekton Result")
```

Comment:
Great~ Thanks for such detailed comments! I learned a lot!
I'm gonna resolve this conversation because we're going to separate the reconcilers, so this misleading error message will no longer exist :)